pr_agent/git_providers/bitbucket_provider.py

```diff
@@ -263,3 +286,89 @@ def publish_labels(self, pr_types: list):
+    def get_issue_comment_user(self, comment):
+        return comment['user']['display_name']
+    
+    def get_issue_created_at(self, issue):
+        return str(issue.created_on)
+    
+    def get_username(self, issue, issue_url):
+        workspace_slug, repo_name, original_issue_numbers = self._parse_issue_url(issue_url)
+        return workspace_slug
+
+    
+    def get_repo_issues(self, repo_obj):
+        return repo_obj._Repository__issues.each()
+
+   
+    def parse_issue_url_and_get_comments(self, issue):
+        workspace_slug, repo_name, original_issue_number = self._parse_issue_url(issue)
+        import requests
+
+        url = f"https://api.bitbucket.org/2.0/repositories/{workspace_slug}/{repo_name}/issues/{original_issue_number}/comments"
+
+        payload = {}
+        headers = {}
+
+        response = requests.request("GET", url, headers=headers, data=payload)
+        return response.json()['values']
+    
+    def parse_issue_url_and_create_comment(self, similar_issues_str, issue_url, original_issue_number):
+        workspace_slug, repo_name, original_issue_number = self._parse_issue_url(issue_url)
+        url = f"https://api.bitbucket.org/2.0/repositories/{workspace_slug}/{repo_name}/issues/{original_issue_number}/comments"
+        payload = json.dumps({
+        "content": {
+            "raw": similar_issues_str
+        }
+        })
+        headers = {
+        'Authorization': f'Bearer {get_settings().get("BITBUCKET.BEARER_TOKEN", None)}',
+        'Content-Type': 'application/json'
+        }
+
+        response = requests.request("POST", url, headers=headers, data=payload)
+
+    def parse_issue_url_and_get_repo_obj(self, issue_url):
+        workspace_slug, repo_name, original_issue_number = self._parse_issue_url(issue_url)
+        return self.bitbucket_client.repositories.get(workspace_slug, repo_name)
+    
+    def get_repo_name_for_indexing(self, repo_obj):
+        return repo_obj._BitbucketBase__data['full_name'].lower().replace('/', '-').replace('_/', '-')
+    
+    def check_if_issue_pull_request(self, issue):
+        return False
+    
+    def get_issue_numbers_from_list(self, issues):
+        # convert str to list'
+        int_list = ast.literal_eval(issues)
+        int_list = [int(x) for x in int_list]
+        for issue_number in int_list:
+            return issue_number
+        
+    def parse_issue_url_and_get_similar_issues(self, issue_url, issue_number_similar):
+        workspace_slug, repo_name, original_issue_number = self._parse_issue_url(issue_url)
+        issue = self.bitbucket_client.repositories.get(workspace_slug, repo_name).issues.get(issue_number_similar)
+        return issue
+
+    def parse_issue_url_and_get_main_issue(self, issue_url):
+        workspace_slug, repo_name, original_issue_number = self._parse_issue_url(issue_url)
+        issue = self.bitbucket_client.repositories.get(workspace_slug, repo_name).issues.get(original_issue_number)
+        return issue
```

Comment:
**Suggestion:** Consider using a helper function to parse the issue URL, as this operation is repeated multiple times in the code. This will make the code cleaner and easier to maintain. [maintainability]